packages/admin/admin/src/common/Dialog.tsx

```diff
@@ -0,0 +1,106 @@
+import { Close } from "@comet/admin-icons";
+import {
+    ComponentsOverrides,
+    css,
+    Dialog as MuiDialog,
+    DialogProps as MuiDialogProps,
+    DialogTitle as MuiDialogTitle,
+    IconButton as MuiIconButton,
+    Theme,
+    useThemeProps,
+} from "@mui/material";
+import { ReactNode } from "react";
+
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+
+export type DialogClassKey = "root" | "iconButton" | "titleWrapper" | "dialogTitle";
+
+export type DialogProps = ThemedComponentBaseProps<{
+    iconButton: typeof MuiIconButton;
+    root: typeof MuiDialog;
+    dialogTitle: typeof MuiDialogTitle;
+    titleWrapper: "div";
+}> & {
+    children?: ReactNode;
+    title?: ReactNode;
+    onClose?: () => void;
```

Comment:
MuiDialogProps already defines onClose with a reason argument, so we shouldn't define it here.